src/controller.ts

```diff
@@ -160,6 +160,22 @@ export default function (
       return paymentService.hasUserAppliedCoupon(user.customerId);
     });
 
+    fastify.put('/apply-coupon', async (req, rep) => {
+      const { uuid } = req.user.payload;
+      const user = await usersService.findUserByUuid(uuid);
+
+      try {
+        const applyCoupon = await paymentService.applyCouponToUser(user.customerId);
+        return applyCoupon;
+      } catch (err) {
+        if (err instanceof CouponAlreadyAppliedError) {
+          return rep.status(403).send({ message: 'Coupon already applied' });
+        } else {
+          return rep.status(500).send({ message: 'Internal server error' });
```

Comment:
Please log the error in case an Internal Error happens, so we can see what is happening.